src/Content/Conversation/Repository/UserDefinedChannel.php

```diff
@@ -221,62 +219,30 @@ public function getMatchingChannelUsers(string $searchtext, string $language, ar
 			return [];
 		}
 
-		$this->insertCheckFullTextSearch($searchtext);
-
-		$uids = [];
-
-		foreach ($this->select(['uid' => array_column($users, 'uid'), 'publish' => true, 'valid' => true]) as $channel) {
-			if (in_array($channel->uid, $uids)) {
-				continue;
-			}
-			if (!empty($channel->circle) && ($channel->circle > 0) && !in_array($channel->uid, $uids)) {
-				if (!$this->inCircle($channel->circle, $channel->uid, $owner_id) && !$this->inCircle($channel->circle, $channel->uid, $reshare_id)) {
-					continue;
-				}
-			}
-			if (!empty($channel->languages) && !in_array($channel->uid, $uids)) {
-				if (!in_array($language, $channel->languages)) {
-					continue;
-				}
-			} elseif (!in_array($language, User::getWantedLanguages($channel->uid))) {
-				continue;
-			}
-			if (!empty($channel->includeTags) && !in_array($channel->uid, $uids)) {
-				if (!$this->inTaglist($channel->includeTags, $tags)) {
-					continue;
-				}
-			}
-			if (!empty($channel->excludeTags) && !in_array($channel->uid, $uids)) {
-				if ($this->inTaglist($channel->excludeTags, $tags)) {
-					continue;
+		return $this->db->prepareThrowawayFullTextSearch($searchtext, function (string $identifier) use ($users, $owner_id, $reshare_id, $language, $tags, $media_type, $searchtext) {
+			$uids = [];
+
+			foreach ($this->select(['uid' => array_column($users, 'uid'), 'publish' => true, 'valid' => true]) as $channel) {
+				if (
```

Comment:
That `if`construct is ... massive. Fun fact: in my company, this would be rejected during the code review because this is hardly to debug.

Reply:
Let me see if I can make it look smaller.